pkg/sentry/ipcutil/ipcutil.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ipcutil defines a set of utilities common to sysvipc mechanisms.
+package ipcutil
+
+import (
+	"gvisor.dev/gvisor/pkg/abi/linux"
+	"gvisor.dev/gvisor/pkg/sentry/fs"
+	"gvisor.dev/gvisor/pkg/sentry/kernel/auth"
```

Comment:
Any reason not to put these functions in sentry/kernel/auth since we are depending on that anyway?

For dependencies sake we should probably not have a base ipcutil package that depends on a more specific package like sentry/kernel/auth

Reply:
There's no specific reason for them to be separate, but I'm not sure if sentry/kernel/auth is the right place for them, as it seems more about defining authentication primitives than providing utility functions. Maybe they can be moved to `sentry/kernel/auth/ipcutil`?

Reply:
Yeah, I guess just ipcutil is fine.